clients/venice-client/src/main/java/com/linkedin/venice/cdc/VeniceCDCIO.java

```diff
@@ -0,0 +1,607 @@
+package com.linkedin.venice.cdc;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+import com.linkedin.davinci.consumer.ChangeEvent;
+import com.linkedin.davinci.consumer.VeniceChangeCoordinate;
+import com.linkedin.davinci.consumer.VeniceChangelogConsumer;
+import com.linkedin.davinci.consumer.VeniceCoordinateOutOfRangeException;
+import com.linkedin.venice.pubsub.api.PubSubMessage;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.annotation.Nullable;
```

Comment:
We have been using this annotation flavor elsewhere in the Venice code base, and it is of course fine to keep using it. But I see that this PR introduces similar annotations from (at least?) two other sources: jetbrains and checker framework.

I will admit that I am not well-versed in the nuances between these various flavors of annotations, though I expect that it is likely reasonable to use just one kind of annotation? Unless there are compelling reasons to adopt a new one, then I would suggest sticking to the javax flavor we have already been using, and not introducing the extra jetbrains and checker framework dependencies.

That being said, I'm happy to get educated about the merits of other annotation dependencies if some significant benefit exists.

Reply:
Checker framework has @NonNull annotation vs @NotNull in jetbrains. Believe the former is used for static analysis vs the later during runtime. @NotNull is required here by the code and recommended by the analyzer as well. Happy to switch if a viable alternative to @NotNull exists.

Reply:
Thanks for the suggestion. Fixed.